scripts/reportportal/rp_manual_reporter/collector.py

```
@@ -0,0 +1,493 @@
+    placeholder_files = scan_placeholder_tests(tests_dir=tests_dir)
+    details: list[PlaceholderTestDetail] = []
+
+    for placeholder_file in placeholder_files:
+        file_path = Path(placeholder_file.file_path)
+        full_path = tests_dir.parent / file_path
+
+        try:
+            source = full_path.read_text(encoding="utf-8")
+            tree = ast.parse(source=source)
+        except (OSError, SyntaxError) as exc:
+            raise RuntimeError(f"Failed to read/parse placeholder file {full_path}: {exc}") from exc
```

Ruff (0.15.18)

[warning] Avoid specifying long messages outside the exception class

(TRY003)